### ISSUE TITLE
The processor eventually fails when unmatching values and types are provided.  

### ISSUE THREAD

**Issue by mgoeminne** [cetic/tsorage#130]
Excerpt from the Processor log:

```
2020-11-02T17:02:21.642150719Z [WARN] [11/02/2020 17:02:21.641] [default-akka.actor.default-dispatcher-8] [RestartWithBackoffFlow(akka://default)] Restarting graph due to failure. stack_trace: 
2020-11-02T17:02:21.642161372Z java.lang.IllegalArgumentException: Expected boolean; got 0
2020-11-02T17:02:21.642164818Z 	at be.cetic.tsorage.data.datatype.BooleanSupport$.fromJson(BooleanSupport.scala:17)
2020-11-02T17:02:21.642168318Z 	at be.cetic.tsorage.data.datatype.BooleanSupport$.fromJson(BooleanSupport.scala:9)
2020-11-02T17:02:21.642171458Z 	at be.cetic.tsorage.data.datatype.DataTypeSupport.asRawUdtValue(DataTypeSupport.scala:113)
2020-11-02T17:02:21.642174529Z 	at be.cetic.tsorage.processor.flow.CassandraWriter$.$anonfun$writeMessageAsync$1(CassandraWriter.scala:39)
2020-11-02T17:02:21.642177611Z 	at scala.collection.immutable.List.map(List.scala:286)
2020-11-02T17:02:21.642180652Z 	at be.cetic.tsorage.processor.flow.CassandraWriter$.writeMessageAsync(CassandraWriter.scala:32)
2020-11-02T17:02:21.642183700Z 	at be.cetic.tsorage.processor.flow.CassandraWriter$.$anonfun$createWriteMsgFlow$1(CassandraWriter.scala:60)
2020-11-02T17:02:21.642186939Z 	at akka.stream.impl.fusing.MapAsyncUnordered$$anon$31.onPush(Ops.scala:1403)
2020-11-02T17:02:21.642189957Z 	at akka.stream.impl.fusing.GraphInterpreter.processPush(GraphInterpreter.scala:541)
2020-11-02T17:02:21.642193068Z 	at akka.stream.impl.fusing.GraphInterpreter.processEvent(GraphInterpreter.scala:495)
2020-11-02T17:02:21.642196038Z 	at akka.stream.impl.fusing.GraphInterpreter.execute(GraphInterpreter.scala:390)
2020-11-02T17:02:21.642199154Z 	at akka.stream.impl.fusing.GraphInterpreterShell.runBatch(ActorGraphInterpreter.scala:624)
2020-11-02T17:02:21.642202206Z 	at akka.stream.impl.fusing.ActorGraphInterpreter$SimpleBoundaryEvent.execute(ActorGraphInterpreter.scala:55)
2020-11-02T17:02:21.642205271Z 	at akka.stream.impl.fusing.ActorGraphInterpreter$SimpleBoundaryEvent.execute$(ActorGraphInterpreter.scala:51)
2020-11-02T17:02:21.642208334Z 	at akka.stream.impl.fusing.ActorGraphInterpreter$BatchingActorInputBoundary$OnNext.execute(ActorGraphInterpreter.scala:94)
2020-11-02T17:02:21.642223856Z 	at akka.stream.impl.fusing.GraphInterpreterShell.processEvent(ActorGraphInterpreter.scala:599)
2020-11-02T17:02:21.642226916Z 	at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$processEvent(ActorGraphInterpreter.scala:768)
2020-11-02T17:02:21.642229773Z 	at akka.stream.impl.fusing.ActorGraphInterpreter$$anonfun$receive$1.applyOrElse(ActorGraphInterpreter.scala:783)
2020-11-02T17:02:21.642232498Z 	at akka.actor.Actor.aroundReceive(Actor.scala:534)
2020-11-02T17:02:21.642235582Z 	at akka.actor.Actor.aroundReceive$(Actor.scala:532)
2020-11-02T17:02:21.642238219Z 	at akka.stream.impl.fusing.ActorGraphInterpreter.aroundReceive(ActorGraphInterpreter.scala:690)
2020-11-02T17:02:21.642240923Z 	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:573)
2020-11-02T17:02:21.642243547Z 	at akka.actor.ActorCell.invoke(ActorCell.scala:543)
2020-11-02T17:02:21.642246129Z 	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:269)
2020-11-02T17:02:21.642248706Z 	at akka.dispatch.Mailbox.run(Mailbox.scala:230)
2020-11-02T17:02:21.642251343Z 	at akka.dispatch.Mailbox.exec(Mailbox.scala:242)
2020-11-02T17:02:21.642253926Z 	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
2020-11-02T17:02:21.642256987Z 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
2020-11-02T17:02:21.642259691Z 	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
2020-11-02T17:02:21.642262334Z 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
2020-11-02T17:02:21.642265076Z 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
2020-11-02T17:02:21.642267865Z 
``` 
There were a series of similar messages. While the processor still worked after this one, it eventually completely stopped. 

Hypothesis: A message with a _tbool_ type and a `0` value has been submitted, which is an error. Because the Boolean support cannot process a `0` value, an exception is raised, which crashes the Processor. Consequently, the message is not committed (it is not successfully written to Cassandra). When the Processor restarts, it reads the message again, which leads to the same result. 

After a while, the process is completely stalled with important lags on the partitions. There is an evidence of this on the Kafka logs:

```
root@tsorage-kafka-0:/# kafka-consumer-groups --bootstrap-server localhost:9092 --group ts_processing --describe

TOPIC           PARTITION  CURRENT-OFFSET  LOG-END-OFFSET  LAG             CONSUMER-ID                                                   HOST            CLIENT-ID
raw             6          910457          1104884         194427          consumer-ts_processing-1-73a66e6f-38e0-41e4-b3b3-109ad89f90f9 /10.44.0.7      consumer-ts_processing-1
raw             18         579031          702597          123566          consumer-ts_processing-1-73a66e6f-38e0-41e4-b3b3-109ad89f90f9 /10.44.0.7      consumer-ts_processing-1
raw             26         567074          688111          121037          consumer-ts_processing-1-73a66e6f-38e0-41e4-b3b3-109ad89f90f9 /10.44.0.7      consumer-ts_processing-1
raw             1          562430          682444          120014          consumer-ts_processing-1-73a66e6f-38e0-41e4-b3b3-109ad89f90f9 /10.44.0.7      consumer-ts_processing-1
raw             12         860425          1044180         183755          consumer-ts_processing-1-73a66e6f-38e0-41e4-b3b3-109ad89f90f9 /10.44.0.7      consumer-ts_processing-1
raw             22         631232          766131          134899          consumer-ts_processing-1-73a66e6f-38e0-41e4-b3b3-109ad89f90f9 /10.44.0.7      consumer-ts_processing-1
raw             2          726710          881795          155085          consumer-ts_processing-1-73a66e6f-38e0-41e4-b3b3-109ad89f90f9 /10.44.0.7      consumer-ts_processing-1
raw             30         564658          685166          120508          consumer-ts_processing-1-73a66e6f-38e0-41e4-b3b3-109ad89f90f9 /10.44.0.7      consumer-ts_processing-1
raw             10         794333          963844          169511          consumer-ts_processing-1-73a66e6f-38e0-41e4-b3b3-109ad89f90f9 /10.44.0.7      consumer-ts_processing-1
raw             16         667162          809540          142378          consumer-ts_processing-1-73a66e6f-38e0-41e4-b3b3-109ad89f90f9 /10.44.0.7      consumer-ts_processing-1
raw             27         676739          821160          144421          consumer-ts_processing-1-73a66e6f-38e0-41e4-b3b3-109ad89f90f9 /10.44.0.7      consumer-ts_processing-1
raw             19         636122          771888          135766          consumer-ts_processing-1-73a66e6f-38e0-41e4-b3b3-109ad89f90f9 /10.44.0.7      consumer-ts_processing-1
raw             8          731388          887460          156072          consumer-ts_processing-1-73a66e6f-38e0-41e4-b3b3-109ad89f90f9 /10.44.0.7      consumer-ts_processing-1
raw             0          37946916        38088799        141883          consumer-ts_processing-1-73a66e6f-38e0-41e4-b3b3-109ad89f90f9 /10.44.0.7      consumer-ts_processing-1
raw             3          534737          648870          114133          consumer-ts_processing-1-73a66e6f-38e0-41e4-b3b3-109ad89f90f9 /10.44.0.7      consumer-ts_processing-1
raw             13         751416          911781          160365          consumer-ts_processing-1-73a66e6f-38e0-41e4-b3b3-109ad89f90f9 /10.44.0.7      consumer-ts_processing-1
raw             31         685809          832172          146363          consumer-ts_processing-1-73a66e6f-38e0-41e4-b3b3-109ad89f90f9 /10.44.0.7      consumer-ts_processing-1
raw             23         722814          877069          154255          consumer-ts_processing-1-73a66e6f-38e0-41e4-b3b3-109ad89f90f9 /10.44.0.7      consumer-ts_processing-1
raw             7          641957          778964          137007          consumer-ts_processing-1-73a66e6f-38e0-41e4-b3b3-109ad89f90f9 /10.44.0.7      consumer-ts_processing-1
raw             17         894145          1084970         190825          consumer-ts_processing-1-73a66e6f-38e0-41e4-b3b3-109ad89f90f9 /10.44.0.7      consumer-ts_processing-1
raw             28         562341          682353          120012          consumer-ts_processing-1-73a66e6f-38e0-41e4-b3b3-109ad89f90f9 /10.44.0.7      consumer-ts_processing-1
raw             20         638579          774867          136288          consumer-ts_processing-1-73a66e6f-38e0-41e4-b3b3-109ad89f90f9 /10.44.0.7      consumer-ts_processing-1
raw             14         557585          676581          118996          consumer-ts_processing-1-73a66e6f-38e0-41e4-b3b3-109ad89f90f9 /10.44.0.7      consumer-ts_processing-1
raw             4          646407          784373          137966          consumer-ts_processing-1-73a66e6f-38e0-41e4-b3b3-109ad89f90f9 /10.44.0.7      consumer-ts_processing-1
raw             24         560883          680584          119701          consumer-ts_processing-1-73a66e6f-38e0-41e4-b3b3-109ad89f90f9 /10.44.0.7      consumer-ts_processing-1
raw             11         479018          581233          102215          consumer-ts_processing-1-73a66e6f-38e0-41e4-b3b3-109ad89f90f9 /10.44.0.7      consumer-ts_processing-1
raw             21         854987          1036763         181776          consumer-ts_processing-1-73a66e6f-38e0-41e4-b3b3-109ad89f90f9 /10.44.0.7      consumer-ts_processing-1
raw             29         598065          725701          127636          consumer-ts_processing-1-73a66e6f-38e0-41e4-b3b3-109ad89f90f9 /10.44.0.7      consumer-ts_processing-1
raw             15         771852          936593          164741          consumer-ts_processing-1-73a66e6f-38e0-41e4-b3b3-109ad89f90f9 /10.44.0.7      consumer-ts_processing-1
raw             5          574206          696745          122539          consumer-ts_processing-1-73a66e6f-38e0-41e4-b3b3-109ad89f90f9 /10.44.0.7      consumer-ts_processing-1
raw             25         850498          1032015         181517          consumer-ts_processing-1-73a66e6f-38e0-41e4-b3b3-109ad89f90f9 /10.44.0.7      consumer-ts_processing-1
raw             9          764846          927931          163085          consumer-ts_processing-1-73a66e6f-38e0-41e4-b3b3-109ad89f90f9 /10.44.0.7      consumer-ts_processing-1
```

**Comment by mgoeminne** [comment]
Suggestion: the processor should filter out values that cannot be handled by the support associated with the specified type. A warning/error message should be produced, and the surviving items in the messages should be written to Cassandra. Eventually, the message should be committed, even if not all its items are processed/written.

**Comment by mgoeminne** [comment]
I would not recommand to extend the BooleanSupport so that 0/1 are considered as valid boolean values.

**Comment by mgoeminne** [comment]
Fixed with 7dfa0b5 (to be published)
